packages/anoncreds/src/formats/LegacyIndyCredentialFormatService.ts

```diff
@@ -0,0 +1,603 @@
+
+    const credentialDefinitionResult = await registry.getCredentialDefinition(
+      agentContext,
+      anonCredsCredential.cred_def_id
+    )
+    if (!credentialDefinitionResult.credentialDefinition) {
+      throw new AriesFrameworkError(
+        `Unable to resolve credential definition ${anonCredsCredential.cred_def_id}: ${credentialDefinitionResult.resolutionMetadata.error} ${credentialDefinitionResult.resolutionMetadata.message}`
+      )
+    }
+
+    const revocationRegistryResult = anonCredsCredential.rev_reg_id
+      ? await registry.getRevocationRegistryDefinition(agentContext, anonCredsCredential.rev_reg_id)
+      : null
+    if (revocationRegistryResult && !revocationRegistryResult.revocationRegistryDefinition) {
+      throw new AriesFrameworkError(
+        `Unable to resolve revocation registry definition ${anonCredsCredential.cred_def_id}: ${credentialDefinitionResult.resolutionMetadata.error} ${credentialDefinitionResult.resolutionMetadata.message}`
+      )
+    }
+
+    if (!credentialRecord.credentialAttributes) {
+      throw new AriesFrameworkError(
+        'Missing credential attributes on credential record. Unable to check credential attributes'
+      )
+    }
+
+    // assert the credential values match the offer values
+    const recordCredentialValues = convertAttributesToCredentialValues(credentialRecord.credentialAttributes)
+    assertCredentialValuesMatch(anonCredsCredential.values, recordCredentialValues)
+
+    const credentialId = await anonCredsHolderService.storeCredential(agentContext, {
+      credentialId: utils.uuid(),
+      credentialRequestMetadata,
+      credential: anonCredsCredential,
+      credentialDefinitionId: credentialDefinitionResult.credentialDefinitionId,
+      credentialDefinition: credentialDefinitionResult.credentialDefinition,
+      revocationRegistry: revocationRegistryResult?.revocationRegistryDefinition
+        ? {
+            definition: revocationRegistryResult.revocationRegistryDefinition,
+            id: revocationRegistryResult.revocationRegistryDefinitionId,
+          }
+        : undefined,
+    })
+
+    // If the credential is revocable, store the revocation identifiers in the credential record
+    if (anonCredsCredential.rev_reg_id) {
+      const credential = await anonCredsHolderService.getCredential(agentContext, { credentialId })
+
+      credentialRecord.metadata.add<AnonCredsCredentialMetadata>(AnonCredsCredentialMetadataKey, {
+        credentialRevocationId: credential.credentialRevocationId,
+        revocationRegistryId: anonCredsCredential.rev_reg_id,
+      })
+    }
+
+    credentialRecord.credentials.push({
+      credentialRecordType: this.credentialRecordType,
+      credentialRecordId: credentialId,
+    })
+  }
+
+  public supportsFormat(format: string): boolean {
+    const supportedFormats = [INDY_CRED_ABSTRACT, INDY_CRED_REQUEST, INDY_CRED_FILTER, INDY_CRED]
```

Comment:
You mean a property supportedFormats?

Yeah could do that. This felt a bit more dynamic (as in you could choose how to parse it). But it would be more in line with what we have in the did and anoncreds services  (supportedMethods / supportedIdentifier).  

It would however change the interface (and I'm just extracting the service here). I'm currently working on a refactor for the proofs module. I'll pick it up as part of that PR